Sources/GRPCCore/Metadata.swift

```diff
@@ -494,15 +494,23 @@ extension Metadata.Value: ExpressibleByArrayLiteral {
 
 extension Metadata: CustomStringConvertible {
   public var description: String {
-    String(describing: self.map({ ($0.key, $0.value) }))
+    var description = "["
```

Comment:
Couple of things:
1. For an empty metadata collection we should print `[:]`, right now we'll print `[]`.
2. We can use map and joined here: `self.map { ... }.joined(separator: ", ")`
3. To quote the keys (and get the correct quoting on values) we should use `String(reflecting:)` (to get the `CustomStringConvertible` value).